### ISSUE TITLE
DetailsList - nested groups: rendering issue

### ISSUE THREAD

**Issue by raphaelstary** [microsoft/fluentui#20825]
### Environment Information

- **Package version(s)**:  8.44.0
- **Browser and OS versions**: (fill this out if relevant)

#### Actual behavior:
https://codepen.io/reactraphael/pen/abLoeGR
1. open the codepen
2. scroll at once to the bottom of the DetailsList
3. DetailsList becomes unusable: flickering, expanding and shrinking, while failing to display items

When you scroll to the bottom of a large DetailsList with nested groups
the list becomes unusable and doesn't show items anymore.
If you scroll down slowly all items from a child group will be shown, but after the last item there is a white block, and the next group will not get rendered/shown. 

You can observe the same issue 
(scrolling to the bottom of the list => list flashing/flickering/expanding/shrinking loop) 
with the first GroupedList example from the docs https://developer.microsoft.com/en-us/fluentui#/controls/web/groupedlist

#### Expected behavior:

A DetailsList with nested groups should just work like the docs example "DetailsList - Large Grouped": 
https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/largegrouped

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: High

Products/sites affected: (if applicable)

**Comment by spmonahan** [comment]
I believe this is due to how row and page heights are calculated for virtualization.

These Github issues appear to be related:

- #20037
- #20978

**Comment by sirdawidd** [comment]
It seems the height is wrongly calculated by default
How ever even I'm hardcode the height of the group it seems not resolve problem
   getGroupHeight={() => 48}
Disabling virtualization helps and resolve the problem , but in my case when I have at least 1k of rows to render list takes on my  pc 4 seconds where with virtualization only 1 sec.
 onShouldVirtualize={() => false}

The issue is even worse when you add custom group header render  (onRenderDetailsHeader), which creates groups with different heights(each group differed height)

This subject is repeating like boomerang, and virtualization was never working in DetailsList and GroupedList

[1]https://github.com/microsoft/fluentui/issues/20037
[2](https://github.com/microsoft/fluentui/issues/5094)
[3](https://github.com/microsoft/fluentui/issues/20978)
[4](https://github.com/microsoft/fluentui/issues/4793)
[5](https://github.com/microsoft/fluentui/issues/4591)
[6](https://github.com/microsoft/fluentui/issues/15527)
....

**Comment by piedras77** [comment]
Hello, we have a main feature that we might need to roll back because of some users experiencing this behavior. Specifically, the list jumps back to the start after scrolling when the user has deeply nested folders. Are there any updates or timelines available?

**Comment by SillyKeith** [comment]
Checking in on the status of this request as our group is also hitting this issue.

**Comment by spmonahan** [comment]
Hello! I've recently merged #24460 which introduces a new version of `GroupedList` called `GroupedListV2`. V2 is an API-compatible reimplementation of `GroupedList` (`GroupedList` being the control that provides nesting for `DetailsList`). While I've tested V2 it is essentially an all-new control and as such is being released as "unstable". This means a couple things:

1. The API is subject to change. I don't expect any changes since it is meant to be compatible with `GroupedList`.
2. Not all features from `GroupedList` are currently implemented. For example there are outstanding issues with ARIA and drag and drop is not implemented.

As such V2 is **not current suitable for use in production**. However, it does fix the rendering issue and I'd appreciate you dropping it into a dev or preview environment to verify that V2 works in your application.

To use `GroupedListV2` with `DetailsList` make sure you're on `@fluentui/react@8.96.0` or later. Then:

```tsx
import { GroupedListV2_unstable as GroupedListV2 } from '@fluentui/react';

// ....

<DetailsList
  groupProps={{
    groupedListAs: GroupedListV2 // new prop that overrides the default GroupedList renderer
  }}
  // ... all other props as before
/>
```

Please try out `GroupedListV2` in your application and if you encounter issues please [file a bug report](https://github.com/microsoft/fluentui/issues/new?assignees=&labels=Type%3A+Bug+%F0%9F%90%9B%2CNeeds%3A+Triage+%F0%9F%94%8D&template=bug-report.yml&title=%5BBug%5D%3A+) and assign it to me.

**Comment by SillyKeith** [comment]
+Alan
I've asked our Dev Alan to give this a go and let you know how it works in our Test/Dev environment.

Keith Kepler
Azure Light Edge | Sr. Program Manager
Office (425) 703-3767
Chat with me via Teams!<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fteams.microsoft.com%2Fl%2Fchat%2F0%2F0%3Fusers%3Dkeith.kepler%40microsoft.com&data=04%7C01%7CKeith.Kepler%40microsoft.com%7C0d0b01881c5e4af61a4608d90f492748%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637557629159128966%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=TquijV%2BP050WBJrXaMk0Hr1g5nWtbU6XeSUYIoDAxPw%3D&reserved=0> | ***@***.******@***.***>
